File: nomad-samples/admin/js/basic-live-channel/live-channel/delete-live-channel.js
import * as prjConstants from "../constants/project-constants.js";
import getLiveChannelInputsIds from "./get-live-channel-inputs-ids.js";
import deleteLiveInput from "../live-input/delete-live-input.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * Delete Live Channel and optionally delete its Live Inputs
 *
 * @param {string} authToken        | Authorization token
 * @param {string} channelId        | The ID of the Live Channel to delete
 * @param {Boolean} deleteInputs    | Flag to delete Live Channel's Live Inputs or not
 */
export default async function deleteLiveChannel(authToken, channelId, deleteInputs)
{
    // If delete Live Inputs then get their IDs
    let inputIds = null;
    if (deleteInputs === true) {
        inputIds = await getLiveChannelInputsIds(authToken, channelId);
    }

    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/liveChannel/${channelId}`, {
        method: "DELETE",
        headers: HEADERS
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {

        // If the Live Channel had Live Inputs
        if (deleteInputs && inputIds && inputIds.length > 0) {
            // Loop deleted Live Channel Live Inputs
            for (let index = 0; index < inputIds.length; index++) {
                // Delete Live Input
                await deleteLiveInput(authToken, inputIds[index]);
            }
        }

        return;
    }

    await apiExceptionHandler(RESPONSE, `Delete Live Channel ${channelId} failed`);
}

File: nomad-samples/admin/py/basic-live-channel-1.1/live_channel/delete_live_channel.py
from constants.project_constants import *
from live_channel.get_live_channel_inputs_ids import *
from live_input.delete_live_input import *
from exceptions.api_exception_handler import *
from live_channel.wait_live_channel_status import *
from live_channel.live_channel_statuses import *

import requests, json

def delete_live_channel(AUTH_TOKEN, CHANNEL_ID, DELETE_INPUTS):
    # If delete Live Inputs then get their IDs
    INPUT_IDS = None
    if (DELETE_INPUTS == True):
        INPUT_IDS = get_live_channel_inputs_ids(AUTH_TOKEN, CHANNEL_ID)


    # Create header for the request
    HEADERS = {
        "Authorization": "Bearer " + AUTH_TOKEN,
        'Content-Type': 'application/json'
    }

    try:
        # Send the request
        RESPONSE = requests.delete(f"{ADMIN_URL}/liveChannel/{CHANNEL_ID}" , headers= HEADERS)

        # If the Live Channel had Live Inputs
        if (DELETE_INPUTS and INPUT_IDS and len(INPUT_IDS) > 0):
            print("Deleting Channel Inputs...")
            # Loop deleted Live Channel Live Inputs
            for ID in INPUT_IDS:
                # Delete Live Input
                delete_live_input(AUTH_TOKEN, ID)



        # Return JSON response
        return RESPONSE.json()

    except:
        api_exception_handler(RESPONSE, f"Delete Live Channel {CHANNEL_ID} failed")

